Samples/custom-transform-adapter/README.md

```diff
@@ -0,0 +1,202 @@
+This will deploy the main Bridge container as well as the adapter container (the parameter `adapter-image` is the name of the adapter image to pull from the registry).
+
+[![Deploy to Azure](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fiot-for-all%2Fiotc-device-bridge%2Fmain%2FSamples%2Fcustom-transform-adapter%2Fazuredeploy.json)
+
+### API surface
+The deployment will forward all requests to `/devices/*` and `/health` directly to the Bridge. All other requests will be routed to the adapter for transformation.
+
+### Uploading configuration file
+A storage account is provisioned with every instance of the Device Bridge. This account will be in the same resource group and contains a File Share named `bridge`.
+By default, the adapter will look for a `config.json` configuration file in the `bridge` File Share. The example command below uploads a configuration file to the Bridge
+storage account. The format of the configuration file is explained in the next sections.
+ 
+`az storage file upload --account-name <account-name> --path ./config.json --share-name bridge --source <path-to-local-config-file>`
+
+Once the configuration file has been uploaded, restart the Bridge instance so the new configuration can be applied. The container logs will display which routes are being configured.
+
+### Logs
+The adapter logs will be published to the same Log Analytics Workspace and the Bridge.
+
+## Configuration
+A configuration file must be in JSON format and have the format below. Each entry of the `d2cMessages` array specifies
+a route that will receive `POST` requests with telemetry messages.
+
+```
+{
+    "d2cMessages": [
+      {
+        // Parameters for route 1
+      },
+      {
+        // Parameters for route 2
+      },
+      // Other routes
+    ]
+}
+```
+
+### Route parameters
+The following route configuration parameters are available:
+
+#### `path`
+Path filter for requests that will be handled by this route. For instance, `"path": "/message"` will handle all `POST` requests made to `/message`.
+A path definition can have parameters. For instance `"path": "/telemetry/{id}"` defined a path parameter `id` and will handle any requests that
+start with `/telemetry/`.
+
+#### `transform`
+[jq](https://stedolan.github.io/jq/) query that defines how request bodies received by this route will be transformed before being forwarded to
+the Bridge. Transformations take the request body as input and must output a JSON object that meets the the Device Bridge
+[telemetry body format](https://github.com/iot-for-all/iotc-device-bridge#device-to-cloud-messages).
+
+For instance, the transformation `"transform": "{ data: . }"` will convert the following request body:
```

Comment:
more of a quality of life thing, but it's pretty painful to write complex JQ queries inside of a JSON string. if there's a reasonable way for the query to be broken out into its own file, that may be nice for users

Reply:
Added. User can now provide transform file path.